### DISCUSSION TITLE
Does Laravel 5.5 supports PHP version 7.2 <= 7.4?

### DISCUSSION THREAD

**Discussion by heyypatrick** [laravel/framework#33996] [General] [answered]
Hi, good day!

We started our project with a PHP version of 7.1. It's all working fine until we upgraded to PHP 7.4. After the upgrade we are having a `count()` related error. I'm aware of it's deprecation but I would like to know if this PHP version supports Laravel 5.5 since it's now EoL? And making the maximum PHP version of Laravel 5.5 is 7.1?

Thanks!

**Details:**
- Laravel 5.5
- PHP 7.4

**Comment by GrahamCampbell** [comment, marked as answer]
Laravel 5.5 LTS supports PHP 7.0-7.3, but **not** PHP 5 or PHP 7.4.

**Reply by GrahamCampbell** [reply]
You should upgrade to Laravel 6 LTS if you need PHP 7.4 support. If you are seeing errors with PHP 7.2 or 7.3, then it's likely you are not using the very latest patch release of 5.5 LTS, or one of your other packages is causing the issue.

**Reply by heyypatrick** [reply]
Alright I already downgraded my PHP version to 7.1. Tho I upgraded to 5.6 with a PHP version 7.3. All good so far but I cannot use 7.4.

**Reply by GrahamCampbell** [reply]
Yep. You should upgrade to Laravel 6 LTS as soon as possible, since Laravel 5.5 LTS goes out of support in a few weeks.

**Reply by heyypatrick** [reply]
Ohh I see, that is noted. And one quick question, this project I am currently working on is dependent on Carbon 1 but seems like on version 5.8 <= 6.*, Carbon 2 is required. Is it possible to use Carbon 1 on Laravel 6.0? It's for a reason that migrating from Carbon 1 to Carbon 2 will require a massive amount of changes on our codebase if I upgrade. Thanks!

**Reply by GrahamCampbell** [reply]
No, soon Laravel 6 LTS will actually require Carbon 2.31.0 or higher, to allow us to bring PHP 8.0 support to Laravel 6 and higher.

**Reply by heyypatrick** [reply]
Okay, I guess we're gonna have some major changes on our codebase. Haha thank you!

**Reply by apolopena** [reply]
@GrahamCampbell Do you know what is the minimum version of Laravel is that supports php 7.4?

**Reply by GrahamCampbell** [reply]
Laravel 6 LTS is the oldest version of Laravel that should be used. It works on all supported versions of PHP (7.3, 7.4 and 8.0).

**Reply by apolopena** [reply]
Thanks I dropped all support of Laravel 5.x in my software.

**Comment by eranthaWELIKALA** [comment]
I had the same issue. The below link explains the issue.

Link - [Warn when counting non-countable types ](https://www.php.net/manual/en/migration72.incompatible.php#migration72.incompatible.warn-on-non-countable-types)
